backend/controllers/user.js

```diff
@@ -23,28 +33,48 @@ const getUserProfile = expressAsyncHandler(async (req, res) => {
 });
 
 // @desc    Update user profile
-// @route   PUT /api/users/profile
+// @route   PUT /api/userprofile
 // @access  Private
 const updateUserProfile = expressAsyncHandler(async (req, res) => {
   const user = await User.findById(req.user._id);
-  if (user) {
-    user.name = req.body.name || user.name;
-    user.email = req.body.email || user.email;
-    if (req.body.password) {
-      user.password = req.body.password;
-    }
-    const updatedUser = await user.save();
-    res.json({
-      _id: updatedUser._id,
-      name: updatedUser.name,
-      email: updatedUser.email,
-      isAdmin: updatedUser.isAdmin,
-      token: generateToken(updatedUser._id),
-    });
-  } else {
+    if (user) {
+        user.photo = req.body.photo || user.photo;
+        user.name = req.body.name || user.name;
+        user.email = req.body.email || user.email;
+        user.username = req.body.username || user.username;
+        user.workTitle = req.body.workTitle || user.workTitle;
+        user.bio = req.body.bio || user.bio;
+        user.toolsAndTech = req.body.toolsAndTech || user.toolsAndTech;
+        if(req.body.socialLinks) {
+          user.socialLinks.linkedIn = req.body.socialLinks.linkedIn || user.socialLinks.linkedIn;
+          user.socialLinks.twitter = req.body.socialLinks.twitter || user.socialLinks.twitter;
+          user.socialLinks.github = req.body.socialLinks.github || user.socialLinks.github;
+          user.socialLinks.medium = req.body.socialLinks.medium || user.socialLinks.medium;
+          user.socialLinks.devTo = req.body.socialLinks.devTo || user.socialLinks.devTo;
+          user.socialLinks.hashnode = req.body.socialLinks.hashnode || user.socialLinks.hashnode;
+          user.socialLinks.leetCode = req.body.socialLinks.leetCode || user.socialLinks.leetCode;
+          user.socialLinks.hackerRank = req.body.socialLinks.hackerRank || user.socialLinks.hackerRank;
+          user.socialLinks.other = req.body.socialLinks.other || user.socialLinks.other;
+        }
+        user.languages = req.body.languages || user.languages;
+        user.experience = req.body.experience || user.experience;
+        user.education = req.body.education || user.education;
+        user.certifications = req.body.certifications || user.certifications;
+        user.achievements = req.body.achievements || user.achievements;
+        user.projects = req.body.projects || user.projects;
+        User.findByIdAndUpdate(
+          { _id: user._id }, { $set: user}, {new: true}, (err, userRes) => {
+            if (err) {
+            return res.status(400).res.json({
```

Comment:
```
res.status(400);
throw new Error("Update Unsuccessful");
```
Change it in this way , as this is the convention followed for throwing errors

Reply:
Changes made. Check it now

Reply:
package-lock.json is still conflicting , once update your branch with latest changes from main

Reply:
Conflicts are now resolved